src/theory/quantifiers/extended_rewrite.cpp

```diff
@@ -1793,15 +1797,75 @@ Node ExtendedRewriter::extendedRewriteStrings(Node ret)
                 }
               }
             }
-            if (isHomogeneous && !std::is_sorted(c[1-i].begin(),c[1-i].end()))
+            if (isHomogeneous)
             {
+              std::sort(c[1 - i].begin(), c[1 - i].end());
```

Comment:
Nit: instead of having `c[1 - i]` and `c[i]` everywhere maybe assign them to variables.

Reply:
I think I'll keep them like this (I dont want an overabundance of variable names).